### PR TITLE
PXB-2544 Exponential Backoff Documentation (8.0)

### PR DESCRIPTION
**Pull request by patrickbirch**
modified:   source/xbcloud/xbcloud.rst
	new file:   source/xbcloud/xbcloud_exbackoff.rst
	new file:   source/xbcloud/xbcloud_gcs.rst
	new file:   source/xbcloud/xbcloud_minio.rst
	new file:   source/xbcloud/xbcloud_s3.rst
	new file:   source/xbcloud/xbcloud_swift.rst

### CODE REVIEW

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 1]
```diff
@@ -46,131 +53,56 @@ as Wasabi or Digital Ocean Spaces, are also supported.
...
-   $ xtrabackup --backup --stream=xbstream --extra-lsndir=/tmp --target-dir=/tmp | \
-   xbcloud put --storage=swift \
-   --swift-container=test \
-   --swift-user=test:tester \
-   --swift-auth-url=http://192.168.8.80:8080/ \
-   --swift-key=testing \
-   --parallel=10 \
-   full_backup
+   xtrabackup --backup --stream=xbstream --incremental-basedir=/tmp \
+   --target-dir=/tmp/inc-backup put [options] <name>
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]:
```suggestion
   xtrabackup --backup --stream=xbstream --incremental-basedir=/storage/backups/ \
   --target-dir=/storage/inc_backup/ | xbcloud put [options] inc_backup
```

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 2]
```diff
@@ -46,131 +53,56 @@ as Wasabi or Digital Ocean Spaces, are also supported.
 Usage
 ================================================================================
 
+The following sample command creates a full backup: 
+
 .. code-block:: bash
 
-   $ xtrabackup --backup --stream=xbstream --target-dir=/tmp | xbcloud \
+   xtrabackup --backup --stream=xbstream --target-dir=/tmp/full-backup | xbcloud \
    put [options] <name>
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]:
```suggestion
 xtrabackup --backup --stream=xbstream --target-dir=/storage/backups/ --extra-lsndir=/storage/backups/ | xbcloud \
 put [options] full_backup
```

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 3]
```diff
@@ -46,131 +53,56 @@ as Wasabi or Digital Ocean Spaces, are also supported.
...
+   xtrabackup --prepare --apply-log-only --target-dir=/tmp/full-backup
 
-.. seealso::
+After the full backup has been prepared, download the incremental backup:
 
-   Cloud Storage Interoperability
-      https://cloud.google.com/storage/docs/interoperability
+.. sourcecode:: bash
+
+   xtrabackup get [options] inc_backup | xbstream -xv -C /tmp/inc-backup
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]:
```suggestion
   xbcloud get [options] inc_backup | xbstream -xv -C /tmp/inc-backup
```

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 4]
```diff
@@ -46,131 +53,56 @@ as Wasabi or Digital Ocean Spaces, are also supported.
...
+
+The downloaded backup is prepared by running the following command:
+
+.. sourcecode:: bash
+
+   xtrabackup --prepare --apply-log-only --target-dir=/tmp --incremental-dir=/tmp/full-backup --incremental-dir=/tmp/inc-backup
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]:
```suggestion
   xtrabackup --prepare --target-dir=/tmp --incremental-dir=/tmp/full-backup --incremental-dir=/tmp/inc-backup
```

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 5]
```diff
@@ -414,107 +288,9 @@ database you can specify only the tables you want to restore:
 This command will download just ``ibdata1`` and ``sakila/payment.ibd`` table
 from the full backup.
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: Where is the command that download this? Is it the one at line 100?

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 1 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: fixed.

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 3 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: Fixed

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 5 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: Yes. Removed line on 415.

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 2 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: Fixed.

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 4 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: Done.

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 6]
```diff
@@ -46,131 +53,56 @@ as Wasabi or Digital Ocean Spaces, are also supported.
 Usage
 ================================================================================
 
+The following sample command creates a full backup: 
+
 .. code-block:: bash
 
-   $ xtrabackup --backup --stream=xbstream --target-dir=/tmp | xbcloud \
+   xtrabackup --backup --stream=xbstream --target-dir=/storage/backups/ --extra-lsndirk=/storage/backups/| xbcloud \
    put [options] <name>
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]:
```suggestion
   put [options] full_backup
```

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst [hunk 7]
```diff
@@ -46,131 +53,56 @@ as Wasabi or Digital Ocean Spaces, are also supported.
...
-      https://cloud.google.com/storage/docs/interoperability
+.. sourcecode:: bash
+
+   xbcloud get [options] inc_backup | xbstream -xv -C /tmp/inc-backup
+
+The downloaded backup is prepared by running the following command:
+
+.. sourcecode:: bash
+
+   xtrabackup --prepare --target-dir=/tmp --incremental-dir=/tmp/full-backup --incremental-dir=/tmp/inc-backup
```
Comment (altmannmarcelo) [inline storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]:
```suggestion
   xtrabackup --prepare --target-dir=/tmp/full-backup --incremental-dir=/tmp/inc-backup
```

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]
two small changes

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 6 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: Done.

File: storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst (hunk 7 above)
Comment (patrickbirch) [reply storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst]: Done.

**Review by altmannmarcelo — APPROVED** [review]

**Pull request merged**